src/aws/nodemanager/nodemanager.py

```diff
@@ -36,6 +54,16 @@ async def run_task_pool(self):
             while True:
                 self.generate_heartbeat()
 
+                # TODO: divide the tasks here. @Karan (see below)
+                """
+                Send them to workers. The available workers are in 
+                self.available_workers. You may need to create a list of divided work where you
+                keep track of the divided work and keep track of tasks that are actuall divided
+                or still waiting for a heartbeat of a worker to give the task.
+                
+                in process_heartbeat you can then actuall send the task to the worker with a
+                CommandPacket.
+                """
```

Comment:
In this function, you should add the logic for which worker which task is given. You could base this on the available instances.

In that same logic, we also need to add resilience for when a worker is stopped/killed.